### PR TITLE
docs(samples): add run details

### PR DESCRIPTION
**Pull request by pmakani**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [X] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-bigquerydatatransfer/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [X] Ensure the tests and linter pass
- [X] Code coverage does not decrease (if any source code was changed)
- [X] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: samples/snippets/src/main/java/com/example/bigquerydatatransfer/RunDetails.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  }
+
+  public static void runDetails(String runId) throws IOException {
+    try (DataTransferServiceClient dataTransferServiceClient = DataTransferServiceClient.create()) {
+      GetTransferRunRequest request = GetTransferRunRequest.newBuilder().setName(runId).build();
+      TransferRun run = dataTransferServiceClient.getTransferRun(request);
+      System.out.print("Run details retrieved successfully :" + run.getName() + "\n");
+    } catch (ApiException ex) {
+      System.out.print("Run details not found." + ex.toString());
+    }
```
Comment (lesv) [inline samples/snippets/src/main/java/com/example/bigquerydatatransfer/RunDetails.java:44]: Ok, but we prefer ([Sample Format guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/SAMPLE_FORMAT.md#exception-handling)) that exceptions not be caught if you aren't going to actually do something with them.

File: samples/snippets/src/main/java/com/example/bigquerydatatransfer/RunDetails.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import java.io.IOException;
+
+// Sample to get run details from transfer config.
+public class RunDetails {
+
+  public static void main(String[] args) throws IOException {
+    // TODO(developer): Replace these variables before running the sample.
+    String runId = "MY_RUN_ID";
+    // `projects/{project_id}/transferConfigs/{config_id}/runs/{run_id}` or
+    // `projects/{project_id}/locations/{location_id}/transferConfigs/{config_id}/runs/{run_id}`
```
Comment (stephaniewang526) [inline samples/snippets/src/main/java/com/example/bigquerydatatransfer/RunDetails.java:33]:
```suggestion
    // runId examples:
    // `projects/{project_id}/transferConfigs/{config_id}/runs/{run_id}` or
    // `projects/{project_id}/locations/{location_id}/transferConfigs/{config_id}/runs/{run_id}`
    String runId = "MY_RUN_ID";
```

File: samples/snippets/src/test/java/com/example/bigquerydatatransfer/RunDetailsIT.java [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    datasetName = "MY_DATASET_NAME_TEST_" + ID;
+    // create a temporary dataset
+    bigquery = BigQueryOptions.getDefaultInstance().getService();
+    bigquery.create(DatasetInfo.of(datasetName));
+    // create a scheduled query
+    String query =
+        "SELECT CURRENT_TIMESTAMP() as current_time, @run_time as intended_run_time, "
+            + "@run_date as intended_run_date, 17 as some_integer";
+    String destinationTableName =
+        "MY_DESTINATION_TABLE_" + UUID.randomUUID().toString().substring(0, 8) + "_{run_date}";
```
Comment (lesv) [inline samples/snippets/src/test/java/com/example/bigquerydatatransfer/RunDetailsIT.java:89]: This is probably ok, but some of those bits are actually constants.  https://tools.ietf.org/html/rfc4122

File: samples/snippets/src/test/java/com/example/bigquerydatatransfer/RunDetailsIT.java [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+            .build();
+    try (DataTransferServiceClient dataTransferServiceClient = DataTransferServiceClient.create()) {
+      ProjectName parent = ProjectName.of(PROJECT_ID);
+      CreateTransferConfigRequest request =
+          CreateTransferConfigRequest.newBuilder()
+              .setParent(parent.toString())
+              .setTransferConfig(transferConfig)
+              .build();
+      name = dataTransferServiceClient.createTransferConfig(request).getName();
+      System.out.println("\nScheduled query created successfully :" + name);
```
Comment (lesv) [inline samples/snippets/src/test/java/com/example/bigquerydatatransfer/RunDetailsIT.java:113]: Hard coding the `\n` will be caught by various static analyzers.

**Review by lesv — APPROVED** [review]
Consider the comments.

**Review by stephaniewang526 — APPROVED** [review]

**Comment by stephaniewang526** [comment]
Reaching out to DTS TL about quota issue: 
``` 
com.google.api.gax.rpc.ResourceExhaustedException: io.grpc.StatusRuntimeException: RESOURCE_EXHAUSTED: Quota exceeded for quota group 'DataTransferServiceOthersGroup' and limit 'ProjectQps' of service 'bigquerydatatransfer.googleapis.com' for consumer 'project_number:779844219229'.
```

will be in touch about this one.

**Comment by stephaniewang526** [comment]
For all the DTS samples that involve dataset and scheduled query, let's use our newly created env vars:
```
BIGQUERYDATATRANSFER_DATASET_NAME
DTS_TRANSFER_CONFIG_NAME
```

**Pull request merged**